rfcs/0016-permissionless-assets-for-ibc-and-cw.md

```diff
@@ -0,0 +1,217 @@
+
+
+```mermaid
+erDiagram
+    gov ||--|| pallet-assets-registry : "root update"
+    pallet-assets-registry ||--|| pallet-assets : "root update"
+    user }|--|| pallet-assets-registry : "permissionless update"
+    ibc-hook ||--|| pallet-assets-registry : "permissionless update"
+    xcm-configuration ||--|| pallet-assets-registry : "permissionless update"
+    xcm-configuration ||--|| pallet-assets : "transfer mint burn"
+    ibc-hook ||--|| pallet-assets : "transfer mint burn"
+    runtime-adapter ||--|| pallet-assets-registry : "permissionless update"
+    runtime-adapter ||--|| pallet-assets : "transfer mint burn"
+    cw-precompile ||--|| pallet-assets-registry : "permissionless update"
+    cw-precompile ||--|| pallet-assets : "transfer mint burn"
+    user }|--|| pallet-assets : "transfer mint burn"
+    gov ||--|| pallet-assets : "transfer mint burn"
+    xcm-channel }|--|| xcm-configuration : "transfer mint burn, permissionless update"
+    ibc-channel }|--|| ibc-hook : "transfer mint burn, permissionless update"    
+    cw-contract }|--|| cw-precompile : "transfer mint burn, permissionless update"    
+    dex }|--|| runtime-adapter : "transfer mint burn, permissionless update"    
+```
+
+### Storage
+
+Storage is more flat than semantic model.
+
+Admin to mint/burn, existential deposits are more related to `assets` pallet. As these are frequent use during mint/burn/transfer inside chain protocols.
+
+Metadata/location/sufficiency/receivability are more lated tp `assets-registry` pallet. These are lower frequency on chain usage outer boundary usage in BYOG/bridges.
+
+### ED (Existential Deposit)
+
+Existential Deposit(ED) is a safeguard against bloating storage of the chain. If a user transfers a nonvaluable asset and sets its ED to 0 or 1, this user will be able to affect performance of the parachain by transferring this asset to many accounts. Thus, to avoid this potential attack vector, we will use Asset Hub (formerly Statemine) approach of non-sufficient assets: a user to be able to recieve permissionless asset needs to hold ED in PICA on Picasso and LAYR on Composable parachains. This approach will limit the growth of the storage. 
+
+The chain needs to make sure that permissionless assets can be transferred to accounts that do not have ED in parachain's native token. It is also required to fail transfers that will result in drop of native token below ED if this account has nonsufficient tokens. [Reference](https://substrate.stackexchange.com/questions/2447/influence-of-existential-deposits-on-account-assets)
+
+
+#### Recieving Account doesnt hold pica during bridging
+**Potential solution**: relayer will send PICA to the reciever in case reciever doesnt have it. In case bridged asset is FlatFee asset, additional fee equal to ED in PICA will be charged from the transferred amount. Otherwise, percentage fee will be applied(To be configured)
```

Comment:
is this included?

Reply:
cc @kkast may you clarify what this means?

Reply:
Ok, I got it. i will rephrase what is now as per IBC protcool.